src/Model/ResultManager.php

```diff
@@ -2,8 +2,48 @@
 
 namespace App\Model;
 
-class ResultManager
+class ResultManager extends AbstractManager
 {
     public const TABLE = 'answer';
-    // select values from answer
+
+    public function fetchValuesByAnswer(array $results): array
+    {
+        // fetching all data from database
+        $answerValues = $this->selectAll();
+
+        // initializing values array
+        $values = [];
+
+        // returning values corresponding to selected answers from the array we got from database
+        foreach ($results as $result) {
+            foreach ($answerValues as $answerValue) {
+                if ($result == $answerValue["label"]) {
+                    array_push($values, $answerValue["answer_value"]);
+                }
+            }
+        }
+        return $values;
+    }
+
+    public function calculateTotalFootprint(array $values): float
+    {
+        // a simple sum of all values we got in the fetchValuesByAnswer function
+        return array_sum($values);
+    }
+
+    public function calculateFootprintByCat(array $values): array
+    {
+        // dividing the values array in smaller arrays (4)
+        // WON'T WORK if questions are added
+        $slicedResults = array_chunk($values, 4, false);
+
+        // initializing footprintByCategory array
+        $footprintByCategory = [];
+
+        // calculating each category and putting the results in an array
+        foreach ($slicedResults as $slicedResult) {
+            array_push($footprintByCategory, array_sum($slicedResult));
+        }
+        return $footprintByCategory;
+    }
```

Comment:
These methods do not belong to the Manager(Model) but to the Controller. Remember that a manager open a connection with the database, and we don't want that if we only do a calculateTotalFootprint().
The best practice should be to create a new classe ResultService, put those two functions in it and call it in your controller.

Reply:
This issue has (theoretically) been fixed in commit [6872172](https://github.com/WildCodeSchool/2022_03_PHP_Lyon_P2_Green_Monkey/pull/6/commits/6872172ad71c639d793b428c64477f8c29442aa5)